ui/packages/consul-nspaces/app/components/consul/nspace/form/index.hbs

```diff
@@ -0,0 +1,169 @@
+<div
+  class="consul-nspace-form"
+  ...attributes
+>
+    <DataWriter
+      @sink={{uri
+        '/${partition}/${nspace}/${dc}/nspace'
+        (hash
+          partition=''
+          nspace=''
+          dc=@item.Datacenter
+        )
+      }}
+      @type={{'nspace'}}
+      @label={{"Namespace"}}
+      @ondelete={{fn (if @ondelete @ondelete @onsubmit) @item}}
+      @onchange={{fn (optional @onsubmit) @item}}
+      as |writer|>
+        <BlockSlot @name="removed" as |after|>
+          <Consul::Nspace::Notifications 
+            {{notification
+              after=(action after)
+            }}
+            @type="remove"
+          />
+        </BlockSlot>
+
+    <BlockSlot @name="content">
+
+{{#let 
+
+  (not (can "write nspaces"))
+
+  @item
+
+  (hash
+    help='Must be a valid DNS hostname. Must contain 1-64 characters (numbers, letters, and hyphens), and must begin with a letter. Once created, this cannot be changed.'
+    Name=(array
+      (hash
+        test='^[a-zA-Z0-9]([a-zA-Z0-9-]{0,62}[a-zA-Z0-9])?$'
+        error='Name must be a valid DNS hostname.'
+      )
+    )
+  )
+
+  (hash
+    Description=(array)
+  )
+
+as |readOnly item Name Description|}}
+<form
+  {{on 'submit' (fn writer.persist item)}}
+  {{disabled readOnly}}
+>
+
+<StateChart 
+  @src={{state-chart 'validate'}}
+as |State Guard ChartAction dispatch state|>
```

Comment:
I mentioned this in another PR, theres an upcoming refactor of StateChart exports/yields to use a single object rather than multiple exports. I want to do that in a separate PR on its own though instead of mixed up in feature work.